src/modules/News/NewsList.tsx

```diff
@@ -0,0 +1,149 @@
+import { News } from '@UG/libs/types';
+import { useEffect, useMemo, useCallback, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { StateType } from 'src/store';
+import { useGetNews } from './hooks';
+import moment from 'moment';
+import { useTranslation } from 'react-i18next';
+
+interface StateProps {
+  isLoading: boolean;
+  newsList: News[];
+  errorMessage: string | null;
+}
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+  color: ${({ theme }) => theme.palette.secondary.dark};
+  font-family: 'Montserrat';
+  font-size: 24px;
+  font-weight: 700;
+`;
+
+export const NewsList = () => {
+  const { t } = useTranslation();
+  const { getNewsList } = useGetNews();
+  const theme = useTheme();
+  const { isLoading, newsList, errorMessage } = useSelector<StateType, StateProps>(state => ({
+    isLoading: state.news.isLoading,
+    newsList: state.news.newsList,
+    errorMessage: state.news.error,
+  }));
+
+  const [activeButton, setActiveButton] = useState('ALL-button');
+
+  const clickedButtonHandler = useCallback(
+    (name: string) => {
+      setActiveButton(name + '-button');
+      getNewsList(name);
+    },
+    [getNewsList],
+  );
+
+  useEffect(() => {
+    getNewsList();
+  }, [getNewsList]);
+
+  const buttons = useMemo(
+    () => (
+      <Box mb={2}>
+        {['ALL', 'INF', 'MFI'].map(name => (
+          <NewsButton
+            name={name}
+            key={name}
+            type="button"
+            text={t('newsPage.' + name.toLowerCase())}
+            className={activeButton == name + '-button' ? 'selected' : ''}
+            onClick={() => clickedButtonHandler(name)}
+          ></NewsButton>
+        ))}
+      </Box>
+    ),
+    [t, activeButton, clickedButtonHandler],
+  );
+
+  const newsTiles: JSX.Element[] = useMemo(
+    () =>
+      newsList.map(({ title, datetime, source, shortBody, photo, _id }) => (
+        <StyledLink to={_id} key={title} data-cy="news-tile-container">
+          <DetailsTile backgroundColor={theme.palette.background.paper}>
+            <Grid container direction="row" justifyContent="flex-start" alignItems="flex-start">
+              <Grid item xs={5} style={{ display: 'flex', alignItems: 'center' }}>
+                <Box
+                  component="img"
+                  sx={{
+                    height: 210,
+                    width: 300,
+                    objectFit: 'cover',
+                    borderRadius: '15px 0px 0px 15px',
+                  }}
+                  src={photo}
+                />
+              </Grid>
+              <Grid
+                item
+                xs={7}
+                style={{
+                  overflow: 'hidden',
+                  textOverflow: 'ellipsis',
+                  display: '-webkit-box',
+                  WebkitLineClamp: '8',
+                  WebkitBoxOrient: 'vertical',
+                }}
+              >
```

Comment:
![image](https://github.com/UniversityOfGdanskProjects/UI-KIOSK/assets/97166296/8b26e32c-a6a2-477c-ae7b-c8b8b5e13676)

At times, trimming the text may not have any sense. Additionally, please remove any unnecessary white spaces

Reply:
In most cases, the issue lies on the MFI and INF side; the text I scrape looks like this in its original form, when it comes to the length of the text.

Reply:
You can use trim(), it removes whitespace on beginning and end of string.

Reply:
done

Reply:
Also remove other white spaces, e.g. \n

Reply:
Please change spaces between title, date and description

Reply:
done